src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,146 @@
+            .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        recommendationRequestRepository.delete(recommendationRequest);
+
+        return genericMessage("RecommendationRequest with id %s deleted".formatted(id));
+    }
+
+    /**
+     * Any admin can update a single Recommendation Request
+     * 
+     * @param id       the id of the Recommendation Request to update
+     * @param incoming the updated Recommendation Request
+     * @return the updated Recommendation Request object
+     */
+    @Operation(summary = "An admin can update a single recommendation request")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("/admin")
+    public RecommendationRequest updateRecommendationRequestAsAdmin(
+        @Parameter(name = "id") @RequestParam Long id,
+        @RequestBody @Valid RecommendationRequest incoming) {
+
+        RecommendationRequest recommendationRequest = 
+            recommendationRequestRepository
+                .findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        recommendationRequest.setProfessorName(incoming.getProfessorName());
+        recommendationRequest.setProfessorEmail(incoming.getProfessorEmail());
+        recommendationRequest.setRequesterName(incoming.getRequesterName());
+        recommendationRequest.setRecommendationTypes(incoming.getRecommendationTypes());
+        recommendationRequest.setDetails(incoming.getDetails());
+        recommendationRequest.setStatus(incoming.getStatus());
+        recommendationRequest.setSubmissionDate(incoming.getSubmissionDate());
+        recommendationRequest.setCompletionDate(incoming.getCompletionDate());
+
+        recommendationRequestRepository.save(recommendationRequest);
+
+        return recommendationRequest;
+    }
+
+    /**
+     * The user who posted a RecommendationRequest can update their RecommendationRequest
+     * 
+     * @param id       the id of the Recommendation Request to update
+     * @param incoming the updated Recommendation Request
+     * @return the updated Recommendation Request object
+     */
+    @Operation(summary = "User can update their RecommendationRequest")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PutMapping("")
+    public RecommendationRequest updateRecommendationRequestAsUser( 
+        @Parameter(name = "id") @RequestParam Long id,
+        @RequestBody @Valid RecommendationRequest incoming) {
+
+        User currentUser = getCurrentUser().getUser(); 
+        RecommendationRequest recommendationRequest = 
+            recommendationRequestRepository
+                .findByIdAndUser(id, currentUser)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        recommendationRequest.setProfessorName(incoming.getProfessorName());
+        recommendationRequest.setProfessorEmail(incoming.getProfessorEmail());
```

Comment:
No... once you set up a request to be for a certain professor, you don't change it to another one!
```suggestion
        
```

if you made it to the wrong professor, you delete it and send a new one to the right one.

Imagine if someone requested a recommendation from me on November 1st, due on December 1st.  Then on Nov 28, they change it to be a recommendation from Prof. Mirza, which shows that it's been on her todo list since November 1st.  I think she's be pretty irritated!